tests/modules/vault-secrets.nix

```diff
@@ -0,0 +1,78 @@
+{ nixosPath, self, ... }@args:
+
+(import "${nixosPath}/tests/make-test-python.nix" ({ pkgs, ... }: rec {
+  name = "vault-secrets";
+  nodes = {
+    server = { pkgs, ... }:
+      let
+        serverArgs =
+          "-dev -dev-root-token-id='root' -dev-listen-address='0.0.0.0:8200'";
+      in {
+        environment.systemPackages = [ pkgs.vault ];
+        # An unsealed dummy vault
+        networking.firewall.allowedTCPPorts = [ 8200 ];
+        systemd.services.dummy-vault = {
+          wantedBy = [ "multi-user.target" ];
+          path = with pkgs; [ getent ];
+          serviceConfig.ExecStart =
+            "${pkgs.vault}/bin/vault server ${serverArgs}";
+        };
+      };
+
+    client = { pkgs, ... }: {
+      imports = [ self.nixosModules.vault-secrets ];
+
+      environment.systemPackages = [ pkgs.vault pkgs.jq ];
+      environment.variables.VAULT_ADDR = "http://server:8200";
+
+      # FIXME: How to provision testing approle credentials?
+      vault-secrets = {
+        vaultAddress = "http://server:8200";
+        secrets.test = { services = [ ]; };
+      };
+    };
+  };
+
+  # API: https://github.com/NixOS/nixpkgs/blob/master/nixos/lib/test-driver/test-driver.py
+  testScript = let
+    sshToServer =
+      "ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i privkey.snakeoil server";
+
+      fakeFlake = {
+        nixosConfigurations.client = self.inputs.nixpkgs.lib.nixosSystem {
+          modules = [ nodes.client ];
+          inherit (pkgs) system;
+        };
+      };
```

Comment:
I have to admit it took me far too long to realize this is not nested under `sshToServer`, but just indented incorrectly.